server/src/unifyfs_server.c

```diff
@@ -248,6 +251,24 @@ static int process_servers_hostfile(const char* hostfile)
     return (int)UNIFYFS_SUCCESS;
 }
 
+
+/* call cleanup for failed clients */
+static void process_cleanup_requests(int num_cleanup,
+                                     arraylist_t* cleanup_arraylist)
+{
+    ABT_mutex_lock(app_configs_abt_sync);
+
+    for (int i = 0; i < num_cleanup; i++) {
+        int* app_id = arraylist_get(cleanup_arraylist, i);
```

Comment:
This code needs to match what you put into the arraylist. Currently it does not.